[3006.x] Update requirements

dwoz opened the pull request:
Name         Version ID                  Fix Versions
------------ ------- ------------------- -------------
jinja2       3.1.4   GHSA-q2x7-8rv6-6q7h 3.1.5
jinja2       3.1.4   GHSA-gmj6-6f8f-6699 3.1.5
urllib3      1.26.18 GHSA-34jh-p97f-mpxf 1.26.19,2.2.2

Fixes #67794
Fixes #67128

dmurphy18 reviewed (dismissed):
Need to fix the lint problem :)

dwoz committed:
Update requirements to account for cves

Name         Version ID                  Fix Versions
------------ ------- ------------------- -------------
jinja2       3.1.4   GHSA-q2x7-8rv6-6q7h 3.1.5
jinja2       3.1.4   GHSA-gmj6-6f8f-6699 3.1.5
urllib3      1.26.18 GHSA-34jh-p97f-mpxf 1.26.19,2.2.2

dwoz committed:
Add changelog for updated dependencies

dwoz commented:
> Looks like there are some deprecations that need to be addressed. https://github.com/saltstack/salt/actions/runs/13448015248/attempts/4?pr=67757
> 
> Deprecated version https://github.com/pyca/pyopenssl/blob/25.0.0/CHANGELOG.rst#2410-2024-03-09

We can not upgrade cryptography because these deprecations will cause us to have to introduce breaking changes. So we'll need to do that in 3008
